### PR TITLE
Feature self reflection

### LINKED ISSUE #17: [User Story Feature] Get introspective questions
**User Story**
As a reflective computing student, I wish to get a set of introspectvie questions on demand so that I can reflect and grow emotionally at my own pace. 

**Is your feature request related to a problem? Please describe.**
Due to hectic and fast paced school work, I always do not have time to reflect on myself. What I have achieved/learned so far in the semester? What skills I wish to obtain in the future? What I have done correctly or what are the areas of improvement? Sometimes I do not know what to start with to reflect on myself. Hence, I wish to have an opportunity to have a set of frequently asked introspective questions and use them to reflect on myself as well. 

**Describe the solution you'd like**
Users can ask WellNUS++ to get a **random** set of introspective questions for users to view.
Command: 
```
get
```
A set of 5 random introspective questions will be generated. For example: 
```
Here’s some questions for you:
1. What’s one thing you are grateful for today?
2. What do I need to change about myself?
3. Am I letting matters that are out of my control stress me out?
.
.
.
N. What is one thing you are proud of today?
```

**Additional context**
At this point, every time users will get a random set of introspective questions. This is for users to get exposed to different questions and possibly reflect on different apsects of themselves. At later stages, users can LIKE a specific question and review it afterwards.

### PR DESCRIPTION
**Pull request by wenxin-c**
Fixes #17 
Set up the main flow of self reflect section and the main method of get reflection questions. 
Things to be done: 
1. JUnit test
2. Integrate exit program method with the main controller
3. Integrate UI with main UI class

### CODE REVIEW

**Comment by nichyjt** [comment]
@wenxin-c can make this a draft PR first as the core boilerplate gets merged into master

File: src/main/java/seedu/duke/reflection/ReturnCommand.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    public ReturnCommand(HashMap<String, String> arguments) throws BadCommandException {
+        super(arguments);
+    }
+
+    /**
+     * To check, this keyword refers to command itself or with a short description?
+     *
+     * @return
+     */
```
Comment (nichyjt) [inline src/main/java/seedu/duke/reflection/ReturnCommand.java:22]: Refer to above comment

File: src/main/java/seedu/duke/reflection/SelfReflection.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import seedu.duke.exception.BadCommandException;
+import seedu.duke.exception.InvalidCommandException;
+
+import java.util.NoSuchElementException;
+
+/**
+ * The main body of self reflect section.
+ */
+public class SelfReflection {
+    private static final String LOGO = "\n"
```
Comment (nichyjt) [inline src/main/java/seedu/duke/reflection/SelfReflection.java:12]: Where is the constructor?

File: src/main/java/seedu/duke/reflection/SelfReflection.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        "Did I make time for myself this week?",
+        "Am I making time for my social life?",
+        "What scares me the most right now?",
+        "What is something I find inspiring?",
+        "What is something that brings me joy?",
+        "When is the last time I gave back to others?",
+        "What matters to me most right now?"};
+
+    private static final ReflectUi UI = new ReflectUi();
+    private static boolean isExit = false;
```
Comment (nichyjt) [inline src/main/java/seedu/duke/reflection/SelfReflection.java:38]: This (`static`) may cause spooky behaviour if we have multiple instances of `SelfReflection`

File: src/main/java/seedu/duke/reflection/ReflectionManager.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    @Override
+    protected void setSupportedFeatureManagers() {
+
+    }
+
+    @Override
+    public void runEventDriver() {
+        SelfReflection selfReflection = new SelfReflection();
+        selfReflection.run();
+    }
```
Comment (nichyjt) [inline src/main/java/seedu/duke/reflection/ReflectionManager.java:77]: Does this compile? There is no constructor for SelfReflection.

File: src/test/java/seedu/duke/reflection/SelfReflectionTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        SelfReflection.setUpQuestions();
+        int fullArrayLength = questions.size();
+        QuestionManager.clearQuestions();
+        assertEquals(0, emptyArrayLength);
+        assertEquals(10, fullArrayLength);
+    }
+}
+
+
+
```
Comment (nichyjt) [inline src/test/java/seedu/duke/reflection/SelfReflectionTest.java:59]: Can remove the extra newlines

**Review by nichyjt — CHANGES_REQUESTED** [review]
For now I've added some comments for you to consider. 

With respect to the difficulties in running `Manager`, this will be patched soon.

File: src/main/java/seedu/duke/reflection/ExitCommand.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        return ARGUMENT;
+    }
+
+    /**
+     * To be completed after main manager is merged.
+     */
+    @Override
+    public void execute() {
+
+    }
```
[the remark this answers is not in the document]
Comment (wenxin-c) [reply src/main/java/seedu/duke/reflection/ExitCommand.java:49]: This exit means exit WellNUS++, return command means go back main app. I havent added this in cause I didnt figure out the variable that controls main app loop.

File: src/main/java/seedu/duke/command/CommandParser.java [hunk 7]
```diff
@@ -146,5 +146,6 @@ public String getMainArgument(String userInput) throws BadCommandException {
         String[] parameters = userInput.split(" ");
         return parameters[0];
     }
-
 }
+
```
[the remark this answers is not in the document]
Comment (wenxin-c) [reply src/main/java/seedu/duke/command/CommandParser.java:150]: is it I typed by accident?? Was it not there originally? but isnt there supposed to be a new line at end of file??

File: src/main/java/seedu/duke/reflection/QuestionManager.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
+package seedu.duke.reflection;
+
+import java.util.ArrayList;
+
+public class QuestionManager {
```
[the remark this answers is not in the document]
Comment (wenxin-c) [reply src/main/java/seedu/duke/reflection/QuestionManager.java:5]: originally they are static attributes/methods for Question, but the class it very long so i factored it out, is it redundant?

File: src/main/java/seedu/duke/reflection/ReflectionManager.java (hunk 4 above)
Comment (wenxin-c) [reply src/main/java/seedu/duke/reflection/ReflectionManager.java:77]: I think it will just auto generate an empty constructor? I'm not too sure but I shall add constructor

File: src/main/java/seedu/duke/reflection/SelfReflection.java (hunk 3 above)
Comment (wenxin-c) [reply src/main/java/seedu/duke/reflection/SelfReflection.java:38]: which static you refer to?

File: src/main/java/seedu/duke/command/CommandParser.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (nichyjt) [reply src/main/java/seedu/duke/command/CommandParser.java:150]: Yes, need just 1 newline.

File: src/main/java/seedu/duke/reflection/SelfReflection.java [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+        {"What are three of my most cherished personal values?",
+        "What is my purpose in life?",
+        "What is my personality type?",
+        "Did I make time for myself this week?",
+        "Am I making time for my social life?",
+        "What scares me the most right now?",
+        "What is something I find inspiring?",
+        "What is something that brings me joy?",
+        "When is the last time I gave back to others?",
+        "What matters to me most right now?"};
```
Comment (nichyjt) [inline src/main/java/seedu/duke/reflection/SelfReflection.java:30]: The coding standard did not define this but based on how these array literals have been defined in the codebase so far, you can consider changing this to the format
```java
QUESTIONS = {
    "What is .....",
    "What is ......",
};
```
for consistency

File: src/main/java/seedu/duke/reflection/ReflectionQuestion.java [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    /**
+     * Covert each reflect question to a string to be printed
+     *
+     * @return question description with its status
+     */
+    @Override
+    public String toString() {
+        return "[" + (isLike ? "LIKE" : " ") + "]" + questionDescription;
+    }
+}
```
Comment (nichyjt) [inline src/main/java/seedu/duke/reflection/ReflectionQuestion.java:55]: Once again, the LIKE feature should not be implemented yet. Refer to https://github.com/AY2223S2-CS2113-T12-4/tp/pull/35#discussion_r1133052273

File: src/main/java/seedu/duke/reflection/SelfReflection.java (hunk 3 above)
Comment (nichyjt) [reply src/main/java/seedu/duke/reflection/SelfReflection.java]: isExit

File: src/main/java/seedu/duke/exception/InvalidCommandException.java [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+/**
+ * Exception caused by invalid command.
+ */
+public class InvalidCommandException extends Exception {
+    public InvalidCommandException(String errorMessage) {
+        super(errorMessage);
+    }
+}
+
+
```
Comment (nichyjt) [inline src/main/java/seedu/duke/exception/InvalidCommandException.java:12]: Why not use `BadCommandException`? They indicate the same category of errors

File: src/main/java/seedu/duke/reflection/InvalidCommand.java [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+     * Throw error and alert users of invalid command.
+     *
+     * @throws InvalidCommandException Invalid command from users
+     */
+    public void alertError() throws InvalidCommandException {
+        throw new InvalidCommandException(INVALID_COMMAND_MESSAGE);
+    }
+}
+
+
```
Comment (nichyjt) [inline src/main/java/seedu/duke/reflection/InvalidCommand.java:21]: Is this file really needed? I don't understand the benefit of having a wrapper here. 
Why cant the offending code `ReflectionManager, line 209` just throw the error direct there?

File: src/main/java/seedu/duke/reflection/QuestionManager.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (nichyjt) [reply src/main/java/seedu/duke/reflection/QuestionManager.java:5]: I don't see a `Question` class, do you mean `SelfReflection`?

When I trace I see its only usage at `SelfReflection` and that file itself is quite short. 
I think this class is not needed - it just adds an extra layer of abstraction that can achieved without this class.

**Review by nichyjt — COMMENTED** [review]
Some comments for your consideration. These are more targetted towards the structure of the code itself and aims to check if the complexity is too high

File: src/main/java/seedu/duke/reflection/GetCommand.java [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+    private static final int NUM_OF_RANDOM_QUESTIONS = 5;
+    private static final String FEATURE_NAME = "Self Reflection";
+    private static final String COMMAND_KEYWORD = "get";
+    private static final String FULL_DESCRIPTION = "";
+    private static final String ARGUMENT = "get";
+    private static final String PAYLOAD = "";
+    private static final int ARGUMENT_PAYLOAD_SIZE = 1;
+    private static final String INVALID_COMMAND_MSG = "Command is invalid.";
+    private static final String INVALID_COMMAND_NOTES = "Please check the available commands "
+            + "and the format of commands.";
```
Comment (YongbinWang) [inline src/main/java/seedu/duke/reflection/GetCommand.java:23]: Should the + sign be underneath the Please?

File: src/main/java/seedu/duke/reflection/ReflectionManager.java [hunk 14]
```diff
@@ -0,0 +1,216 @@
...
+    private static final String NO_ELEMENT_MESSAGE = "There is no new line of input, please key in inputs.";
+    private static final String INVALID_COMMAND_MESSAGE = "Please enter a valid command.";
+    private static final String EMPTY_COMMAND_MESSAGE = "The command is empty, please check your input.";
+    private static final ReflectUi UI = new ReflectUi();
+
+    // I need to set this as static if I want to set it to true if ExitCommand object.
+    // If I create another object in ExitCommand, the corresponding isExit will be for a new object,
+    // not the one we intend to terminate.
+    // Anyone has other ideas??
+    private static boolean isExit;
```
Comment (YongbinWang) [inline src/main/java/seedu/duke/reflection/ReflectionManager.java:29]: Can check the way i implemented my exit function in atomic habits using an instance of exit command to check.

File: src/main/java/seedu/duke/reflection/ReflectionManager.java [hunk 15]
```diff
@@ -0,0 +1,216 @@
...
+     * Commands available at this moment are:
+     * <li>Get a random set of reflection questions<br/>
+     * <li>Return back main interface<br/>
+     * <li>Exit program<br/>
+     *
+     * @throws BadCommandException Empty command
+     */
+    public void execute() throws BadCommandException, InvalidCommandException {
+        switch (commandType) {
+        case GET_COMMAND:
```
Comment (YongbinWang) [inline src/main/java/seedu/duke/reflection/ReflectionManager.java:195]: Can consider changing the function name to executeCommand()

File: src/main/java/seedu/duke/reflection/QuestionManager.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (wenxin-c) [reply src/main/java/seedu/duke/reflection/QuestionManager.java:5]: Sorry I was referring to class ```ReflectionQuestion```. The main loop was originally inside ```SelfReflection``` class, now i moved it to ```ReflectionManager``` so I became pretty short

File: src/main/java/seedu/duke/reflection/GetCommand.java (hunk 13 above)
Comment (wenxin-c) [reply src/main/java/seedu/duke/reflection/GetCommand.java:23]: I think it's 8 spaces from the previous line, @haoyangw can help double check this?

File: src/main/java/seedu/duke/reflection/QuestionManager.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (wenxin-c) [reply src/main/java/seedu/duke/reflection/QuestionManager.java:5]: I merged ```QuestionManager``` class with ```SelfReflection``` class

File: src/main/java/seedu/duke/reflection/GetCommand.java (hunk 13 above)
Comment (YongbinWang) [reply src/main/java/seedu/duke/reflection/GetCommand.java:22]: yes, confirmed just now 8 spaces from previous line

File: src/main/java/seedu/duke/reflection/ReflectionManager.java (hunk 14 above)
Comment (wenxin-c) [reply src/main/java/seedu/duke/reflection/ReflectionManager.java:28]: I leave it like this first, if there is any spooky behaviours I'll try ur method:D

File: src/main/java/seedu/duke/reflection/ReflectionManager.java [hunk 16]
```diff
@@ -0,0 +1,203 @@
...
+
+    /**
+     * Decide which command to create based on command type.<br/>
+     * <br/>
+     * Commands available at this moment are:
+     * <li>Get a random set of reflection questions<br/>
+     * <li>Return back main interface<br/>
+     * <li>Exit program<br/>
+     *
+     * @throws BadCommandException Empty command
```
Comment (haoyangw) [inline src/main/java/seedu/duke/reflection/ReflectionManager.java:182]: Same as above: I believe 'If an empty command was given' would be better

File: src/main/java/seedu/duke/reflection/ReflectionQuestion.java [hunk 17]
```diff
@@ -0,0 +1,35 @@
+package seedu.duke.reflection;
+
+/**
+ * ReflectQuestion class is used to create reflect question objects.<br/>
+ * <br/>
+ * The static attribute questions is used to keep track all the questions created. <br/>
+ * Attribute isLike is to keep track of whether a user likes this question or not.
```
Comment (haoyangw) [inline src/main/java/seedu/duke/reflection/ReflectionQuestion.java:7]: Personal opinion is to describe the attributes as a comment directly above the attribute's declaration, not in the class-level JavaDoc. These attributes are private after all; The actual variable name should be kept abstract and not revealed.

File: src/main/java/seedu/duke/reflection/ReflectionQuestion.java [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+     * @return Question description
+     */
+    public String getQuestionDescription() {
+        return this.questionDescription;
+    }
+
+    /**
+     * Covert each reflect question to a string to be printed
+     *
+     * @return question description with its status
```
Comment (haoyangw) [inline src/main/java/seedu/duke/reflection/ReflectionQuestion.java:28]: Capitalise 'q' of `question`

File: src/main/java/seedu/duke/reflection/ReflectionQuestion.java [hunk 19]
```diff
@@ -0,0 +1,35 @@
...
+     * Get the description of question.
+     *
+     * @return Question description
+     */
+    public String getQuestionDescription() {
+        return this.questionDescription;
+    }
+
+    /**
+     * Covert each reflect question to a string to be printed
```
Comment (haoyangw) [inline src/main/java/seedu/duke/reflection/ReflectionQuestion.java:26]: Spelling error? Is it 'Convert' instead of 'Covert'?

File: src/main/java/seedu/duke/reflection/SelfReflection.java [hunk 20]
```diff
@@ -0,0 +1,89 @@
...
+            + " \\___ \\|  __| | |    |  __|   |  _  /|  __| |  __| | |    |  __|| |       | |    | || |  | | . ` |\n"
+            + " ____) | |____| |____| |      | | \\ \\| |____| |    | |____| |___| |____   | |   _| || |__| | |\\  |\n"
+            + "|_____/|______|______|_|      |_|  \\_\\______|_|    |______|______\\_____|  |_|  "
+            + "|_____\\____/|_| \\_|\n";
+    private static final String GREETING_MESSAGE = "Welcome to WellNUS++ Self Reflection section:D"
+            + System.lineSeparator() + "Feel very occupied and cannot find time to self reflect?"
+            + System.lineSeparator() + "No worries, this section will give you the opportunity to reflect "
+            + "and improve on yourself!!";
+
+    // Questions are adopted from website: https://www.usa.edu/blog/self-discovery-questions/
```
Comment (haoyangw) [inline src/main/java/seedu/duke/reflection/SelfReflection.java:22]: Good to link the source!

File: src/main/java/seedu/duke/reflection/SelfReflection.java [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+     */
+    public static void clearQuestions() {
+        questions.clear();
+    }
+
+    /**
+     * Add new reflect question object into the list.<br/>
+     * This method is more for developer usage at this stage as under are not allowed to add their own questions. <br/>
+     *
+     * @param question newly created reflect question to be added into the list
```
Comment (haoyangw) [inline src/main/java/seedu/duke/reflection/SelfReflection.java:83]: Capitalise 'n' of `newly`

File: src/test/java/seedu/duke/reflection/SelfReflectionTest.java [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+    @Test
+    void setUpQuestions_checkArrayLength_success() {
+        SelfReflection.clearQuestions();
+        ArrayList<ReflectionQuestion> questions = SelfReflection.getQuestions();
+        int emptyArrayLength = questions.size();
+        SelfReflection selfReflection = new SelfReflection();
+        int fullArrayLength = questions.size();
+        SelfReflection.clearQuestions();
+        assertEquals(0, emptyArrayLength);
+        assertEquals(10, fullArrayLength);
```
Comment (haoyangw) [inline src/test/java/seedu/duke/reflection/SelfReflectionTest.java:45]: Not client code I admit, but still good to avoid magic literals in my opinion

File: src/test/java/seedu/duke/reflection/ReturnCommandTest.java [hunk 23]
```diff
@@ -0,0 +1,49 @@
...
+    // Test whether wrong format command exception is caught or not.
+    @Test
+    void execute_checkWrongCmdFormat_expectFalse() throws BadCommandException {
+        ReflectionManager reflectionManager = new ReflectionManager();
+        reflectionManager.setArgumentPayload(RETURN_COMMAND_WRONG_FORMAT);
+        HashMap<String, String> returnArgumentPayload = reflectionManager.getArgumentPayload();
+        ReturnCommand returnCmd = new ReturnCommand(returnArgumentPayload);
+        returnCmd.execute();
+        ReflectionManager newReflectionManager = new ReflectionManager();
+        assertEquals(false, newReflectionManager.getIsExit());
```
Comment (haoyangw) [inline src/test/java/seedu/duke/reflection/ReturnCommandTest.java:46]: Not client code, but still good to avoid magic literals(`false`) in my opinion

File: src/test/java/seedu/duke/reflection/ReturnCommandTest.java [hunk 24]
```diff
@@ -0,0 +1,49 @@
...
+    // Test whether isExit is false for a new ReflectionManager object after exiting from previous one.
+    @Test
+    void execute_checkNewObjectIsExit_expectFalse() throws BadCommandException {
+        ReflectionManager reflectionManager = new ReflectionManager();
+        reflectionManager.setArgumentPayload(RETURN_COMMAND);
+        HashMap<String, String> returnArgumentPayload = reflectionManager.getArgumentPayload();
+        ReturnCommand returnCmd = new ReturnCommand(returnArgumentPayload);
+        returnCmd.execute();
+        ReflectionManager newReflectionManager = new ReflectionManager();
+        assertEquals(false, newReflectionManager.getIsExit());
```
Comment (haoyangw) [inline src/test/java/seedu/duke/reflection/ReturnCommandTest.java:34]: Not client code, but still good to avoid magic literals(`false`) in my opinion

File: src/test/java/seedu/duke/reflection/ReturnCommandTest.java [hunk 25]
```diff
@@ -0,0 +1,49 @@
...
+
+    // Test whether ReturnCommand execute() method can terminate self reflection or not.
+    @Test
+    void execute_checkIsExit_expectTrue() throws BadCommandException {
+        ReflectionManager reflectionManager = new ReflectionManager();
+        reflectionManager.setArgumentPayload(RETURN_COMMAND);
+        HashMap<String, String> returnArgumentPayload = reflectionManager.getArgumentPayload();
+        ReturnCommand returnCmd = new ReturnCommand(returnArgumentPayload);
+        returnCmd.execute();
+        assertEquals(true, reflectionManager.getIsExit());
```
Comment (haoyangw) [inline src/test/java/seedu/duke/reflection/ReturnCommandTest.java:22]: Not client code, but still good to avoid magic literals(`true`) in my opinion

**Review by haoyangw — CHANGES_REQUESTED** [review]
Code appears to function as intended! There're some pesky code style violations though, fix them and it's LGTM! Great work as always :)

File: src/main/java/seedu/duke/reflection/ReflectUi.java [hunk 26]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    /**
+     * Print greeting logo.
+     *
+     * @param logo Greeting logo
+     */
+    public void printLogo(String logo) {
+        printSeparator();
+        System.out.print(logo);
```
[the remark this answers is not in the document]
Comment (wenxin-c) [reply src/main/java/seedu/duke/reflection/ReflectUi.java:25]: i create a method for ```System.out.print(logo)``` then call this method instead, is this what you want?

File: src/main/java/seedu/duke/reflection/ExitCommand.java [hunk 27]
```diff
@@ -0,0 +1,98 @@
...
+     * @return Command: exit
+     */
+    @Override
+    protected String getCommandKeyword() {
+        return COMMAND_KEYWORD;
+    }
+
+    /**
+     * Get detailed description of an exit command.
+     * FULL_DESCRIPTION is not completed yet.
```
[the remark this answers is not in the document]
Comment (nichyjt) [reply src/main/java/seedu/duke/reflection/ExitCommand.java:37]: The IDE highlights the comment that has TODO at the start. Its meant as a visual aid for you to keep track of what needs to be done

File: src/main/java/seedu/duke/reflection/SelfReflection.java (hunk 20 above)
Comment (wenxin-c) [reply src/main/java/seedu/duke/reflection/SelfReflection.java:22]: u mean links have a specific format in JavaDoc?

File: src/main/java/seedu/duke/reflection/ExitCommand.java (hunk 27 above)
[the remark this answers is not in the document]
Comment (haoyangw) [reply src/main/java/seedu/duke/reflection/ExitCommand.java:37]: Yeah what I meant is what @nichyjt said

File: src/main/java/seedu/duke/reflection/ReflectUi.java (hunk 26 above)
[the remark this answers is not in the document]
Comment (haoyangw) [reply src/main/java/seedu/duke/reflection/ReflectUi.java:25]: Oh ok this is complicated that'll be an unnecessary abstraction instead. Let's just leave it as is first, but think of another way meanwhile. Might get flagged during PE.

[the remark this answers is not in the document]
Comment (haoyangw) [reply src/main/java/seedu/duke/reflection/ReflectUi.java:25]: @nichyjt @YongbinWang @BernardLesley any ideas?

[the remark this answers is not in the document]
Comment (nichyjt) [reply src/main/java/seedu/duke/reflection/ReflectUi.java:25]: A key reason why we decided to have the `TextUi` class is to abstract away `System.out.*` method calls. 

I think that if a method exists to abstract away a certain function (such as printing a message), we should use that method rather than reinvent the wheel. 

Because of that, I think that the call to `System.out` here is in violation of SLAP and makes the code less cohesive.

@wenxin-c Is there no way to print the logo using existing methods within the superclass?

[the remark this answers is not in the document]
Comment (wenxin-c) [reply src/main/java/seedu/duke/reflection/ReflectUi.java:25]: but I tot it's okay to add stuff in subclasses?? cause some behaviours might varry across subclasses?

**Review by haoyangw — APPROVED** [review]
LGTM! No more code style issues here, thanks so much for taking the time to clean up your code!

File: src/main/java/wellnus/reflection/ReflectionManager.java [hunk 28]
```diff
@@ -0,0 +1,195 @@
...
+    private static final String EXIT_COMMAND = "exit";
+    private static final String EXIT_PAYLOAD = "";
+    private static final String NO_ELEMENT_MESSAGE = "There is no new line of input, please key in inputs.";
+    private static final String INVALID_COMMAND_MESSAGE = "Please check the available commands "
+            + "and enter a valid command.";
+    private static final ReflectUi UI = new ReflectUi();
+
+    // I need to set this as static if I want to set it to true if ExitCommand object.
+    // If I create another object in ExitCommand, the corresponding isExit will be for a new object,
+    // not the one we intend to terminate.
```
Comment (nichyjt) [inline src/main/java/wellnus/reflection/ReflectionManager.java:26]: Comments should not be in [first person](https://nus-cs2113-ay2223s2.github.io/website/schedule/week7/topics.html#implementation-code-quality-comments-basic-write-to-the-reader). Write to the reader

**Review by nichyjt — CHANGES_REQUESTED** [review]
Javadoc nit. Sorry, but one of your javadocs was written in first person which is a clear violation of coding standards.

**Review by nichyjt — APPROVED** [review]
Thanks for the hard work. LGTM!
